src/mon/MonClient.cc

```diff
@@ -687,21 +688,27 @@ MonConnection& MonClient::_add_conn(unsigned rank, uint64_t global_id)
 
 void MonClient::_add_conns(uint64_t global_id)
 {
-  uint16_t min_priority = std::numeric_limits<uint16_t>::max();
+  map<uint16_t, vector<unsigned>> rank_by_priority;
   for (const auto& m : monmap.mon_info) {
-    if (m.second.priority < min_priority) {
-      min_priority = m.second.priority;
-    }
+    rank_by_priority[m.second.priority].push_back(monmap.get_rank(m.first));
   }
   vector<unsigned> ranks;
-  for (const auto& m : monmap.mon_info) {
-    if (m.second.priority == min_priority) {
-      ranks.push_back(monmap.get_rank(m.first));
+  if (!rank_by_priority.empty()) {
```

Comment:
we will in a bigger trouble if `rank_by_priority.empty()`. could you instead
```c++
assert(!rank_by_priority.empty());
```
?

Reply:
seems the original code logic should respect an empty monmap. I am not sure this will cause any other potential issue. Updated as you wish, though.